addon/utils/get-index.js

```diff
@@ -0,0 +1,14 @@
+import isObject from '../utils/is-object';
+
+export default function getIndex(array, currentValue) {
+  let index;
+  if (isObject(currentValue)) {
+    index = array.findIndex((object) => {
+      return JSON.stringify(object) === JSON.stringify(currentValue);
```

Comment:
could we pass in an option to this function, say `deepEqual`? It can default to `false`, then we only need to compare the reference using `===` and stringify if set to `true`